docs/PARALLEL_CHECKSUMS.md

```diff
@@ -0,0 +1,32 @@
+# PARALLEL CHECKSUMS
+
+## sha2-256-parallel-8
+
+### 0xb518
+
+This variant of sha2-256 is designed to enable very large files
+to be uploaded and hashed in parallel using fixed size blocks
+(8MiB to start with), with the final result being a "top_hash"
+that is an order-independent hash of the individual hashes.
```

Comment:
"order-independent" sounds wrong to me. It's true that it doesn't depend on the upload order, but it certainly depends on the order of chunks within the file, so I'd make that more clear.